app/src/main/java/com/unc0ded/shopdeliver/views/fragments/vendorInventoryFragment.java

```diff
@@ -71,27 +70,30 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
                     .show();
         });
 
-        listenerRegistration = db.collection("Inventory").whereEqualTo("vendorId", Objects.requireNonNull(vendorAuth.getCurrentUser()).getUid())
-                .addSnapshotListener(((value, error) -> {
-                    if (error != null)
-                        Log.e("Failed", error.toString());
-                    else {
-                        int size = inventoryArray.size();
-                        for (int i = 0; i < size; i++) {
-                            inventoryArray.remove(0);
+        if (vendorAuth.getUid() != null){
+            listenerRegistration = db.collection("Inventory").whereEqualTo("vendorId", Objects.requireNonNull(vendorAuth.getCurrentUser()).getUid())
+                    .addSnapshotListener(((value, error) -> {
+                        if (error != null)
+                            Log.e("Failed", error.toString());
+                        else {
+                            int size = inventoryArray.size();
+                            for (int i = 0; i < size; i++) {
+                                inventoryArray.remove(0);
+                            }
+                            for (QueryDocumentSnapshot documentSnapshot: value) {
+                                inventoryArray.add(new Gson().toJsonTree(documentSnapshot.getData()).getAsJsonObject());
+                            }
+                            adapter.notifyDataSetChanged();
                         }
-                        for (QueryDocumentSnapshot documentSnapshot: value) {
-                            inventoryArray.add(new Gson().toJsonTree(documentSnapshot.getData()).getAsJsonObject());
-                        }
-                        adapter.notifyDataSetChanged();
-                    }
-                }));
+                    }));
+        }
     }
```

Comment:
Is this part of the migration left? Because I did see an empty VendorMainActvityViewModel, and no repo for Vendor.

Reply:
Yep I'll reduce that np